pkgs/development/python-modules/kaggle/default.nix

```diff
@@ -0,0 +1,53 @@
+{ buildPythonPackage
+, certifi
+, fetchPypi
+, lib
+, python-dateutil
+, python-slugify
+, six
+, requests
+, tqdm
+, urllib3
+}:
+
+buildPythonPackage rec {
+  pname = "kaggle";
+  version = "1.5.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0f5qrkgklcpgbwncrif7aw4f86dychqplh7k3f4rljwnr9yhjb1w";
+  };
+
+  # The version bounds in the setup.py file are unnecessarily restrictive.
+  patchPhase = ''
+    substituteInPlace setup.py \
+      --replace 'urllib3 >= 1.21.1, < 1.25' 'urllib3'
+  '';
+
+  propagatedBuildInputs = [
+    certifi
+    python-dateutil
+    python-slugify
+    requests
+    six
+    tqdm
+    urllib3
+  ];
+
+  # Tests try to access the network.
+  checkPhase = ''
+    export HOME="$TMP"
+    mkdir -p "$HOME/.kaggle/"
+    echo '{"username":"foobar","key":"00000000000000000000000000000000"}' > "$HOME/.kaggle/kaggle.json"
+    $out/bin/kaggle --help > /dev/null
+  '';
+  pythonImportsCheck = [ "kaggle" ];
```

Comment:
@jonringer Thanks for the tip!

Unfortunately, `kaggle` requires the file `~/.kaggle/kaggle.json` to exist, otherwise the CLI throws an exception and ends the process with an error code.

In order to make `kaggle --help` not end with an error code, I had to create fake api credentials.  This is somewhat hacky, but still appears to work for simple things that don't involve actually hitting the API (like just calling `--help`).

I've confirmed this builds, so if this looks alright, I'd be happy to go ahead and merge this in.

Reply:
I think the warning if fine, the `$out/bin/kaggle --help` still returns 0, even with the warnings.

My main concern is just that a dependency bump map break it, and we won't know about it until runtime (after someone does an update and the package then fails)